packages/url/init.lua

```diff
@@ -46,8 +46,11 @@ function package.declareSettings (_)
 end
 
 function package:registerCommands ()
-
   self:registerCommand("href", function (options, content)
+    if (not content or #content == 0 and not content[1]) and options.src then
```

Comment:
This going to break I think, if content is not an AST table but (for instance) a function - e.g. when this is used from La code as:
```
SILE.call("href", { src = "..." }, function ()
   -- Some logic formatting the content
end)
```
This would now break with  `attempt to get length of 'content' (a function value)`, I'm afraid.
We already support
- `\href[src=URL]{Some content}`
- `\href{URL}
Not sure we should try something to `\href[src=URL]` (which I guess is the idea here), but be conscious its a bit tricky.

Reply:
Anyway, if we go for it, then use `SU.hasContent(content)` - it's the way to go I guess.

Reply:
@Omikhleia Are you OK with me dropping it? It's too complex to keep I  think and is a holdover from when I was still trying to put the feature into `\url` and not `\href`.

Reply:
Sorry for the long delay. No problem with me for dropping `\href[src=...]` without arg. As I said, I wasn't even sure it was needed in the first place - And indeed you are right, this is a different topic. It will be cool enough to get the hook so one can have, say, color-bracketed URLs in upright font :)